sky/skylet/providers/gcp/config.py

```diff
@@ -506,10 +507,50 @@ def _configure_subnet(config, compute):
     ):
         return config
 
-    subnets = _list_subnets(config, compute)
+    subnets = _list_subnets(config, compute, filter=f'(name="{SKYPILOT_VPC_NAME}")')
```

Comment:
Instead of always using the `SKYPILOT_VPC_NAME`, should we use the original way as before and use this creation as a backup:
1. Check all subnets and use the first possible subnet if any subnet exists
2. If no subnets exist, try to create a `SKYPILOT_VPC_NAME`

Reason:
1. Backward compatibility for any existing user, as a cluster launched with the default VPC will be stopped and relaunched if we use a different subnet, IIUC.
2. To avoid asking all the users to have permission to create a subnet, i.e. the proposal above will only require the user who does not have any subnets available to have that permission.